src/components/Button/ButtonGroup.tsx

```diff
@@ -0,0 +1,116 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import { UIComponent, childrenExist, customPropTypes } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import { Extendable, ItemShorthand, ReactChildren } from '../../../types/utils'
+import Button from './Button'
+
+export interface IButtonGroupProps {
+  as?: any
+  children?: ReactChildren
+  circular?: boolean
+  className?: string
+  content?: React.ReactNode
+  buttons?: ItemShorthand[]
+  styles?: IComponentPartStylesInput
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A button group.
+ */
+class ButtonGroup extends UIComponent<Extendable<IButtonGroupProps>, any> {
+  public static displayName = 'ButtonGroup'
+
+  public static className = 'ui-button__buttons'
```

Comment:
seems that `BEM` conventions are violated here - this reads as `'buttons' element inside 'ui-button' - but I would rather expect something simplier like 'ui-buttons' or 'ui-button-group'`

Reply:
We have utility function for generating the className inside getComponentInfo.ts: 

  info.componentClassName = (info.isChild
    ? \`ui-${info.parentDisplayName}__${info.subcomponentName.replace(
        /Group$/,
        \`${info.parentDisplayName}s\`,
      )}\`
    : \`ui-${info.displayName}\`
  ).toLowerCase()

I was just following it in order to make the component conformant. Do you think we should change this logic?

Reply:
the logic of this function seems correct to me - the problem is that for `info.parentDisplayName` 'button' is used for the case of `ButtonGroup`, and this is semantically incorrect

Reply:
In the end, we decided that the components that contain Group inside the name should result into the main components class with appended 's' in the end (For the ButtonGroup it should be ui-buttons). We are doing this because we already had some logic for the Group components,  and we can still keep the group components together with the main components that they represent.